server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -832,7 +832,7 @@ boolean isGroupSetting() {
          * @return the raw list of dependencies for this setting
          */
         public Set<AffixSettingDependency> getDependencies() {
-            return Collections.unmodifiableSet(dependencies);
+            return dependencies;
```

Comment:
Cleaning this up here because I found it while tracking this bug down and it's too trivial for a separate PR, `this.dependencies` is always immutable so no point in wrapping again.